web3/providers/rpc.py

```diff
@@ -57,6 +74,94 @@ def make_request(self, method, params):
         return response_body
 
 
+class KeepAliveRPCProvider(BaseProvider):
```

Comment:
I think it might be prudent to adjust `RPCProvider` to be a subclass of the `KeepAliveRPCProvider` that raises a [`DeprecationWarning`](https://docs.python.org/2/library/exceptions.html#exceptions.DeprecationWarning) at instantiation.  I've been meaning to start being nicer about breaking changes and now is as good a time as any.